clippy_lints/src/methods.rs

```diff
@@ -623,6 +623,29 @@ declare_lint! {
     "using `as_ref` where the types before and after the call are the same"
 }
 
+
+/// **What it does:** Checks for using `fold` when a more succint alternative exists.
+/// Specifically, this checks for `fold`s which could be replaced by `any`, `all`,
+/// `sum` or `product`.
+///
+/// **Why is this bad?** Readability.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// let _ = (0..3).fold(false, |acc, x| acc || x > 2);
+/// ```
+/// This could be written as:
```

Comment:
I'm pretty sure it's good style to put an empty line above and below code blocks in markdown.

Reply:
The exist lints in this file don't, for whatever that's worth